### ISSUE TITLE
handling of float labels is inconsistent

### ISSUE THREAD

**Issue by gdementen** [liam2/larray#292]
```python
>>> arr = LArray([1, 2, 3], [[1.0, 2.0, 3.0]])
>>> arr
{0}  1.0  2.0  3.0
       1    2    3
>>> # OK
>>> arr[1.0:2.0]
{0}  1.0  2.0
       1    2
>>> # OK
>>> arr[1.0]
1
>>> # this should not work
>>> arr[1:2]
{0}  1.0  2.0
       1    2
>>> # ... like it does for single value
>>> arr[1]
ValueError: 1 is not a valid label for any axis
>>> # maybe in this case we want to have a specific error message like:
>>> arr[1]
ValueError: 1 is not a valid label for any axis, but 1.0 exist maybe you wanted to use that
```

**Comment by gdementen** [comment]
On second thought, since 1 == 1.0, it seems logical that arr[1] == arr[1.0], so maybe we should scrap the whole dtype checking code in Axis.translate?

**Comment by gdementen** [comment]
Initially, I think it was meant to avoid having a[True] == a[1] (or a[False]) because it cause a strange bug somewhere. I don't remember the details though.

**Comment by gdementen** [comment]
IIRC, the idea behind the type checking code was also a way to make guessing the axis faster by failing early in case of incompatible type. That is, if you have string, int and float axes, and you look for a key 10, it only tries the int axes. I am unsure it actually works.

Honestly, I can see some merit in allowing a[1] to target float labels (because sometimes, we are lazy and do not add the .0 for floats and for most of Python it just works). Targeting an int axis with a float on the other hand seems messy. 

In short, I am unsure whether to allow this, but either way, we should be consistent, which is not the case now.
